### PR TITLE
Add pull request template with release note stub

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

Adds a pull request template that includes a release note field and issue fixes

### CODE REVIEW

**Review by trishankatdatadog — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Please fill in the fields below to submit a pull request.  The more information that is provided, the better.
+
+Fixes #Issue:
+Release Notes:
+
+**Description of the changes being introduced by the pull request**:
+
```
[the remark this answers is not in the document]
Comment (asraa) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: that's great!

**Review by trishankatdatadog — DISMISSED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+Please fill in the fields below to submit a pull request.  The more information that is provided, the better.
+
+Fixes #Issue:
+Release Notes:
```
Comment (joshuagl) [inline .github/PULL_REQUEST_TEMPLATE.md:4]:
```suggestion
Release Notes: <!-- What comments/remarks should we include in the release notes for this change? -->
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+Please fill in the fields below to submit a pull request.  The more information that is provided, the better.
+
+Fixes #Issue:
```
Comment (joshuagl) [inline .github/PULL_REQUEST_TEMPLATE.md:3]: For GitHub to automatically close the issue we need to use a specific format see [using keywords in issues and pull requests](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests).
```suggestion
Fixes: #<Issue>
```
?

**Review by joshuagl — COMMENTED** [review]
Thank you for working on this, some minor nits. Let's get this merged! :-)

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (asraa) [reply .github/PULL_REQUEST_TEMPLATE.md:3]: doesn't look like it needs the ":", so removing that

**Review by joshuagl — APPROVED** [review]
👍

**Review by mnm678 — APPROVED** [review]

**Pull request merged**